src/SoftwareApplication.ts

```diff
@@ -20,5 +23,14 @@ export default class SoftwareApplication extends CreativeWork {
   applicationSubCategories: Array<Text | URL> = []
 
   @property('schema:operatingSystem', 'list')
-  operatingSystems: Array<Text> = []
+  operatingSystems: Array<OperatingSystem> = []
+
+  @property('schema:programmingLanguage', 'list')
+  programmingLanguages: Array<ComputerLanguage> = []
```

Comment:
I'm worried that we are diverging from schema.org unnecessarily. `programminLanguage`, `runtimePlatform` and `codeRepository` all exist on [`SoftwareSourceCode`](https://schema.org/SoftwareSourceCode) not `SoftwareApplication`. Can we instead use `SoftwareSourceCode.targetProduct` to link back to the application (or have a new `SoftwareApplication.codes` (or something) with an array of `SoftwareSourceCode`). Ping @apawlik